nginx-controller/controller/controller.go

```diff
@@ -1067,3 +1095,90 @@ func (lbc *LoadBalancerController) ValidateSecret(secret *api_v1.Secret) error {
 
 	return fmt.Errorf("Secret is not a TLS or JWK secret")
 }
+
+// getAllIngress returns a list of all ingress resources
+func (lbc *LoadBalancerController) getAllIngress() ([]nginx.IngressEx, error) {
```

Comment:
I suggest to change this function to return all minions for a particular host. We can also validate Ingress resources here. We can rename it to `getMinionsForHost`.